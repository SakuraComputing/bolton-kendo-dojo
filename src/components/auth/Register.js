import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import TextFieldGroup from '../../common/TextFieldGroup';
import { registerUser } from '../../actions/authActions';

export class Register extends Component {

    state = {
        name: '',
        email: '',
        password: '',
        password2: '',
        errors: {}
    };

    onChange = (e) => {
        this.setState({[e.target.name]: e.target.value})
    };

    onSubmit = (e) => {
        e.preventDefault();
        const newUser = {
            name: this.state.name,
            email: this.state.email,
            password: this.state.password,
            password2: this.state.password2
        };
        this.props.registerUser(newUser, this.props.history);
    };

    render() {
        const { errors } = this.state;
        return (
            <div className="register">
                <div className="container">
                    <h1 className="form-title">Sign Up</h1>
                    <p className="form-text">Create your Members Account</p>
                    <form noValidate onSubmit={this.onSubmit}>
                        <TextFieldGroup
                            placeholder="Name"
                            name="name"
                            value={this.state.name}
                            type="name"
                            onChange={this.onChange}
                            error={errors.name}
                        />
                        <TextFieldGroup
                            placeholder="Email Address"
                            name="email"
                            value={this.state.email}
                            type="email"
                            onChange={this.onChange}
                            error={errors.email}
                        />
                        <TextFieldGroup
                            placeholder="Password"
                            name="password"
                            value={this.state.password}
                            type="password"
                            onChange={this.onChange}
                            error={errors.password}
                        />
                        <TextFieldGroup
                            placeholder="Confirm Password"
                            name="password2"
                            value={this.state.password2}
                            type="password"
                            onChange={this.onChange}
                            error={errors.password2}
                        />

                        <input type="submit" className="button button-info"/>
                    </form>
                </div>
            </div>
        );
    }
}

Register.propTypes = {
    registerUser: PropTypes.func.isRequired,
    auth: PropTypes.object.isRequired,
    errors: PropTypes.object.isRequired
};

const mapStateToProps = (state) => ({
    auth: state.auth,
    errors: state.errors
});

export default connect(mapStateToProps, { registerUser })(withRouter(Register));
