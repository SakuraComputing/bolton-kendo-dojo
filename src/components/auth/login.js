import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { loginUser } from '../../actions/authActions';
import PropTypes from 'prop-types';
import TextFieldGroup from '../../common/TextFieldGroup';

export class Login extends Component {

    constructor() {
        super();
        this.state = {
             email: '',
            password: '',
            errors: {}
        }
    }
    
    componentWillReceiveProps(nextProps) {
        console.log('Component Will Mount', nextProps);
        if(nextProps.errors) {
            this.setState({ errors: nextProps.errors });
        }
    }

    onChange = (e) => {
        this.setState({ [e.target.name]: e.target.value })
    };

    onSubmit = (e) => {
        e.preventDefault();
        const userData = {
            email: this.state.email,
            password: this.state.password
        };
        this.props.loginUser(userData);
    }

    render() {

    const { errors } = this.state;

    return (
      <div>
        <div className="login">
            <div className="content-container">
                <h1 className="form-title">Log In</h1>
                <p className="form-text">Sign in to the Members Area</p>
                <form onSubmit={this.onSubmit}>
                    <TextFieldGroup
                        placeholder="Email Address"
                        name="email"
                        value={this.state.email}
                        type="email"
                        onChange={this.onChange}
                        error={errors.email}
                    />
                    <TextFieldGroup
                        placeholder="Password"
                        name="password"
                        value={this.state.password}
                        type="password"
                        onChange={this.onChange}
                        error={errors.password}
                    />
                    <input
                        type="submit"
                        className="button button-info"
                    />
                </form>
                <div className="login-request">
                    <Link to="/register">New - Click here to request membership</Link>
                </div>
            </div>
        </div>
      </div>
    )
  }
}

Login.propTypes = {
    loginUser: PropTypes.func.isRequired,
    auth: PropTypes.object.isRequired,
    errors: PropTypes.object.isRequired
};

const mapStateToProps = (state) => ({
   auth: state.auth,
   errors: state.errors,
});

export default connect(mapStateToProps, { loginUser })(Login);
