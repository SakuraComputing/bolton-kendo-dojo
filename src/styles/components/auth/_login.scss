.login {
    height: 100vh;
}

.login-request {
    font-size: 110%;
    padding-top: 2em;
}

