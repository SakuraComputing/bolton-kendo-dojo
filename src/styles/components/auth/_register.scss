.register {
    height: 100vh;
    text-align: center;
}