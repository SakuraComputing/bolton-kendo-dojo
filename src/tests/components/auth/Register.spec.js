import React from 'react';
import { Register } from '../../../components/auth/Register';
import { shallow } from 'enzyme';

describe('Register Form', () => {

    let wrapper;

    beforeEach(() => {
        const registerUser = jest.fn();
        const errors = { errors: 'error' }
        const props = { auth: 'auth'}
        wrapper = shallow(<Register
            registerUser={registerUser}
            auth={props}
            errors={errors}
        />);
        console.error = jest.fn();
    });

    afterEach(() => {
        console.error('you cant see me');
        console.error.mockClear();
    });

    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('should set the state on change', () => {
        wrapper.find('TextFieldGroup').at(0).simulate('change', {
            target: { value: 'Change Function' }
        });
        expect(wrapper).toMatchSnapshot();
    });
});