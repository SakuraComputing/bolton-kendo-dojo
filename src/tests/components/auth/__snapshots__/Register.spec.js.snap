// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Form should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register
    auth={
      Object {
        "auth": "auth",
      }
    }
    errors={
      Object {
        "errors": "error",
      }
    }
    registerUser={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <h1
          className="display-4 text-center"
        >
          Sign Up
        </h1>
        <h3
          className="lead text-center"
        >
          Create your Members Account
        </h3>
        <form
          noValidate={true}
          onSubmit={[Function]}
        >
          <TextFieldGroup
            name="name"
            onChange={[Function]}
            placeholder="Name"
            type="name"
            value=""
          />
          <TextFieldGroup
            name="email"
            onChange={[Function]}
            placeholder="Email Address"
            type="email"
            value=""
          />
          <TextFieldGroup
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
          <TextFieldGroup
            name="password2"
            onChange={[Function]}
            placeholder="Confirm Password"
            type="password"
            value=""
          />
          <input
            className="button button-info"
            type="submit"
          />
        </form>
      </div>,
      "className": "register",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="display-4 text-center"
          >
            Sign Up
          </h1>,
          <h3
            className="lead text-center"
          >
            Create your Members Account
          </h3>,
          <form
            noValidate={true}
            onSubmit={[Function]}
          >
            <TextFieldGroup
              name="name"
              onChange={[Function]}
              placeholder="Name"
              type="name"
              value=""
            />
            <TextFieldGroup
              name="email"
              onChange={[Function]}
              placeholder="Email Address"
              type="email"
              value=""
            />
            <TextFieldGroup
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
            <TextFieldGroup
              name="password2"
              onChange={[Function]}
              placeholder="Confirm Password"
              type="password"
              value=""
            />
            <input
              className="button button-info"
              type="submit"
            />
          </form>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up",
            "className": "display-4 text-center",
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Create your Members Account",
            "className": "lead text-center",
          },
          "ref": null,
          "rendered": "Create your Members Account",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TextFieldGroup
                name="name"
                onChange={[Function]}
                placeholder="Name"
                type="name"
                value=""
              />,
              <TextFieldGroup
                name="email"
                onChange={[Function]}
                placeholder="Email Address"
                type="email"
                value=""
              />,
              <TextFieldGroup
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />,
              <TextFieldGroup
                name="password2"
                onChange={[Function]}
                placeholder="Confirm Password"
                type="password"
                value=""
              />,
              <input
                className="button button-info"
                type="submit"
              />,
            ],
            "noValidate": true,
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "error": undefined,
                "name": "name",
                "onChange": [Function],
                "placeholder": "Name",
                "type": "name",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "error": undefined,
                "name": "email",
                "onChange": [Function],
                "placeholder": "Email Address",
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "error": undefined,
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "error": undefined,
                "name": "password2",
                "onChange": [Function],
                "placeholder": "Confirm Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "button button-info",
                "type": "submit",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <h1
            className="display-4 text-center"
          >
            Sign Up
          </h1>
          <h3
            className="lead text-center"
          >
            Create your Members Account
          </h3>
          <form
            noValidate={true}
            onSubmit={[Function]}
          >
            <TextFieldGroup
              name="name"
              onChange={[Function]}
              placeholder="Name"
              type="name"
              value=""
            />
            <TextFieldGroup
              name="email"
              onChange={[Function]}
              placeholder="Email Address"
              type="email"
              value=""
            />
            <TextFieldGroup
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
            <TextFieldGroup
              name="password2"
              onChange={[Function]}
              placeholder="Confirm Password"
              type="password"
              value=""
            />
            <input
              className="button button-info"
              type="submit"
            />
          </form>
        </div>,
        "className": "register",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="display-4 text-center"
            >
              Sign Up
            </h1>,
            <h3
              className="lead text-center"
            >
              Create your Members Account
            </h3>,
            <form
              noValidate={true}
              onSubmit={[Function]}
            >
              <TextFieldGroup
                name="name"
                onChange={[Function]}
                placeholder="Name"
                type="name"
                value=""
              />
              <TextFieldGroup
                name="email"
                onChange={[Function]}
                placeholder="Email Address"
                type="email"
                value=""
              />
              <TextFieldGroup
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />
              <TextFieldGroup
                name="password2"
                onChange={[Function]}
                placeholder="Confirm Password"
                type="password"
                value=""
              />
              <input
                className="button button-info"
                type="submit"
              />
            </form>,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up",
              "className": "display-4 text-center",
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Create your Members Account",
              "className": "lead text-center",
            },
            "ref": null,
            "rendered": "Create your Members Account",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <TextFieldGroup
                  name="name"
                  onChange={[Function]}
                  placeholder="Name"
                  type="name"
                  value=""
                />,
                <TextFieldGroup
                  name="email"
                  onChange={[Function]}
                  placeholder="Email Address"
                  type="email"
                  value=""
                />,
                <TextFieldGroup
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
                />,
                <TextFieldGroup
                  name="password2"
                  onChange={[Function]}
                  placeholder="Confirm Password"
                  type="password"
                  value=""
                />,
                <input
                  className="button button-info"
                  type="submit"
                />,
              ],
              "noValidate": true,
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "error": undefined,
                  "name": "name",
                  "onChange": [Function],
                  "placeholder": "Name",
                  "type": "name",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "error": undefined,
                  "name": "email",
                  "onChange": [Function],
                  "placeholder": "Email Address",
                  "type": "email",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "error": undefined,
                  "name": "password",
                  "onChange": [Function],
                  "placeholder": "Password",
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "error": undefined,
                  "name": "password2",
                  "onChange": [Function],
                  "placeholder": "Confirm Password",
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "button button-info",
                  "type": "submit",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "form",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Register Form should set the state on change 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register
    auth={
      Object {
        "auth": "auth",
      }
    }
    errors={
      Object {
        "errors": "error",
      }
    }
    registerUser={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <h1
          className="display-4 text-center"
        >
          Sign Up
        </h1>
        <h3
          className="lead text-center"
        >
          Create your Members Account
        </h3>
        <form
          noValidate={true}
          onSubmit={[Function]}
        >
          <TextFieldGroup
            name="name"
            onChange={[Function]}
            placeholder="Name"
            type="name"
            value=""
          />
          <TextFieldGroup
            name="email"
            onChange={[Function]}
            placeholder="Email Address"
            type="email"
            value=""
          />
          <TextFieldGroup
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
          <TextFieldGroup
            name="password2"
            onChange={[Function]}
            placeholder="Confirm Password"
            type="password"
            value=""
          />
          <input
            className="button button-info"
            type="submit"
          />
        </form>
      </div>,
      "className": "register",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="display-4 text-center"
          >
            Sign Up
          </h1>,
          <h3
            className="lead text-center"
          >
            Create your Members Account
          </h3>,
          <form
            noValidate={true}
            onSubmit={[Function]}
          >
            <TextFieldGroup
              name="name"
              onChange={[Function]}
              placeholder="Name"
              type="name"
              value=""
            />
            <TextFieldGroup
              name="email"
              onChange={[Function]}
              placeholder="Email Address"
              type="email"
              value=""
            />
            <TextFieldGroup
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
            <TextFieldGroup
              name="password2"
              onChange={[Function]}
              placeholder="Confirm Password"
              type="password"
              value=""
            />
            <input
              className="button button-info"
              type="submit"
            />
          </form>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up",
            "className": "display-4 text-center",
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Create your Members Account",
            "className": "lead text-center",
          },
          "ref": null,
          "rendered": "Create your Members Account",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TextFieldGroup
                name="name"
                onChange={[Function]}
                placeholder="Name"
                type="name"
                value=""
              />,
              <TextFieldGroup
                name="email"
                onChange={[Function]}
                placeholder="Email Address"
                type="email"
                value=""
              />,
              <TextFieldGroup
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />,
              <TextFieldGroup
                name="password2"
                onChange={[Function]}
                placeholder="Confirm Password"
                type="password"
                value=""
              />,
              <input
                className="button button-info"
                type="submit"
              />,
            ],
            "noValidate": true,
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "error": undefined,
                "name": "name",
                "onChange": [Function],
                "placeholder": "Name",
                "type": "name",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "error": undefined,
                "name": "email",
                "onChange": [Function],
                "placeholder": "Email Address",
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "error": undefined,
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "error": undefined,
                "name": "password2",
                "onChange": [Function],
                "placeholder": "Confirm Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "button button-info",
                "type": "submit",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <h1
            className="display-4 text-center"
          >
            Sign Up
          </h1>
          <h3
            className="lead text-center"
          >
            Create your Members Account
          </h3>
          <form
            noValidate={true}
            onSubmit={[Function]}
          >
            <TextFieldGroup
              name="name"
              onChange={[Function]}
              placeholder="Name"
              type="name"
              value=""
            />
            <TextFieldGroup
              name="email"
              onChange={[Function]}
              placeholder="Email Address"
              type="email"
              value=""
            />
            <TextFieldGroup
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
            <TextFieldGroup
              name="password2"
              onChange={[Function]}
              placeholder="Confirm Password"
              type="password"
              value=""
            />
            <input
              className="button button-info"
              type="submit"
            />
          </form>
        </div>,
        "className": "register",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="display-4 text-center"
            >
              Sign Up
            </h1>,
            <h3
              className="lead text-center"
            >
              Create your Members Account
            </h3>,
            <form
              noValidate={true}
              onSubmit={[Function]}
            >
              <TextFieldGroup
                name="name"
                onChange={[Function]}
                placeholder="Name"
                type="name"
                value=""
              />
              <TextFieldGroup
                name="email"
                onChange={[Function]}
                placeholder="Email Address"
                type="email"
                value=""
              />
              <TextFieldGroup
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />
              <TextFieldGroup
                name="password2"
                onChange={[Function]}
                placeholder="Confirm Password"
                type="password"
                value=""
              />
              <input
                className="button button-info"
                type="submit"
              />
            </form>,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up",
              "className": "display-4 text-center",
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Create your Members Account",
              "className": "lead text-center",
            },
            "ref": null,
            "rendered": "Create your Members Account",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <TextFieldGroup
                  name="name"
                  onChange={[Function]}
                  placeholder="Name"
                  type="name"
                  value=""
                />,
                <TextFieldGroup
                  name="email"
                  onChange={[Function]}
                  placeholder="Email Address"
                  type="email"
                  value=""
                />,
                <TextFieldGroup
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
                />,
                <TextFieldGroup
                  name="password2"
                  onChange={[Function]}
                  placeholder="Confirm Password"
                  type="password"
                  value=""
                />,
                <input
                  className="button button-info"
                  type="submit"
                />,
              ],
              "noValidate": true,
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "error": undefined,
                  "name": "name",
                  "onChange": [Function],
                  "placeholder": "Name",
                  "type": "name",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "error": undefined,
                  "name": "email",
                  "onChange": [Function],
                  "placeholder": "Email Address",
                  "type": "email",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "error": undefined,
                  "name": "password",
                  "onChange": [Function],
                  "placeholder": "Password",
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "error": undefined,
                  "name": "password2",
                  "onChange": [Function],
                  "placeholder": "Confirm Password",
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "button button-info",
                  "type": "submit",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "form",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
